### PR TITLE
Web3 signer functionality

### PR DESCRIPTION
**Pull request by TomasArrachea**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -129,8 +137,10 @@ alloy-transport-ws = { version = "0.1", default-features = false }
 alloy-transport-ipc = { version = "0.1", default-features = false }
 alloy-pubsub = { version = "0.1", default-features = false }
 alloy-json-rpc = { version = "0.1", default-features = false }
-alloy-rpc-client = { version = "0.1", default-features = false }
+alloy-rpc-client = "0.2"
 alloy-contract = { version = "0.1", default-features = false }
+alloy-signer-aws = "0.1.0"
```
Comment (pablodeymo) [inline Cargo.toml]: this version should be "0.1"

File: Cargo.toml [hunk 2]
```diff
@@ -129,8 +137,10 @@ alloy-transport-ws = { version = "0.1", default-features = false }
 alloy-transport-ipc = { version = "0.1", default-features = false }
 alloy-pubsub = { version = "0.1", default-features = false }
 alloy-json-rpc = { version = "0.1", default-features = false }
-alloy-rpc-client = { version = "0.1", default-features = false }
+alloy-rpc-client = "0.2"
```
Comment (pablodeymo) [inline Cargo.toml]: we should use version 0.1

File: crates/signer/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+aws-config.workspace = true
+aws-sdk-kms.workspace = true
+eth-keystore.workspace = true
+hex-literal.workspace = true
+thiserror.workspace = true
+tokio.workspace = true
+serde.workspace = true
+url.workspace = true
+testcontainers.workspace = true
+alloy-node-bindings.workspace = true
```
Comment (pablodeymo) [inline crates/signer/Cargo.toml:24]: this sould be placed in dev-dependencies section

File: crates/signer/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+alloy-rpc-client.workspace = true
+aws-config.workspace = true
+aws-sdk-kms.workspace = true
+eth-keystore.workspace = true
+hex-literal.workspace = true
+thiserror.workspace = true
+tokio.workspace = true
+serde.workspace = true
+url.workspace = true
+testcontainers.workspace = true
```
Comment (pablodeymo) [inline crates/signer/Cargo.toml]: this crate should be placed inside dev-dependencies section also

File: crates/signer/src/signer.rs [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+    use alloy_consensus::{SignableTransaction, TxLegacy};
+    use alloy_network::{TxSigner, TxSignerSync};
+    use alloy_node_bindings::Anvil;
+    use alloy_primitives::hex_literal::hex;
+    use alloy_primitives::{address, bytes, keccak256, Address, U256};
+    use alloy_signer::Signature;
+    use alloy_signer_local::PrivateKeySigner;
+    use aws_config::{BehaviorVersion, Region, SdkConfig};
+    use aws_sdk_kms::types::KeyMetadata;
+    use aws_sdk_kms::{
```
Comment (pablodeymo) [inline crates/signer/src/signer.rs]: join this 2 includes inside the second one

File: crates/signer/src/signer.rs [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+        self,
+        config::{Credentials, SharedCredentialsProvider},
+    };
+    use std::str::FromStr;
+    use testcontainers::runners::AsyncRunner;
+    use testcontainers::{
+        core::{IntoContainerPort, WaitFor},
+        GenericImage,
+    };
+    use testcontainers::{ContainerAsync, ImageExt};
```
Comment (pablodeymo) [inline crates/signer/src/signer.rs]: the same here

File: crates/signer/src/signer.rs [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+        assert_eq!(signature, expected_signature);
+    }
+
+    #[tokio::test]
+    async fn sign_transaction_with_aws_signer() {
+        // Start the container running Localstack
+        let _container = start_localstack_container().await;
+
+        let localstack_endpoint = "http://localhost:".to_string() + &LOCALSTACK_PORT.to_string();
```
Comment (pablodeymo) [inline crates/signer/src/signer.rs]:
```suggestion
        let localstack_endpoint = format!("http://localhost:{}", LOCALSTACK_PORT);
```

File: Cargo.toml (hunk 2 above)
Comment (ricomateo) [reply Cargo.toml]: it works if we use the version "0.1" but we have to set `default-features = true`

File: Cargo.toml (hunk 1 above)
Comment (ricomateo) [reply Cargo.toml]: Done!

File: crates/signer/Cargo.toml (hunk 4 above)
Comment (ricomateo) [reply crates/signer/Cargo.toml]: Done

File: crates/signer/src/signer.rs (hunk 5 above)
Comment (ricomateo) [reply crates/signer/src/signer.rs]: Done

File: crates/signer/src/signer.rs (hunk 6 above)
Comment (ricomateo) [reply crates/signer/src/signer.rs]: Done

File: crates/signer/src/signer.rs (hunk 7 above)
Comment (ricomateo) [reply crates/signer/src/signer.rs]: Done

**Review by pablodeymo — APPROVED** [review]
LGTM

**Pull request closed without merging**